### PR TITLE
Search functionalities

### LINKED ISSUE #7967: Search for text with braces in pattern.
When i try to search in source (ctrl+shift+f) wth braces in pattern i find nothing 
ex: `parameters = Object.assign({node: null}, parameters);`
Then i must escape all braces, which is very annoying, to find what i need.
ex: `parameters = Object.assign\({node: null}, parameters\);`

I propose to search with text pattern by default (like in old versions). Regex search pattern should be an option.

### PR DESCRIPTION
**Pull request by charanjit-singh**
Fixes #7967 

### Summary of Changes

* Added Regex, Case Sensitive, Whole word modifiers in search ( Ctrl + shift + F )
* Added styles for active classes on modifier buttons.

### Test Plan

Tests can be done by searching using combination of all modifiers and checking their results.

Example test plan:

- [x] Open test file in browser.
- [x] Press Ctrl + Shift + F for searching.
- [x] In Input box, type the words to test.
- [x] Check/ uncheck the modifiers.
- [x] Verify the expected results.

### Screenshots
screenshot from 2019-02-26 17-24-15
---

screenshot from 2019-02-26 17-24-23

---

screenshot from 2019-02-26 17-24-44

### CODE REVIEW

**Comment by jasonLaster** [comment]
This might be a bit of scope creep, but I think it would be nice if the project search bar were now on the bottom and a similar look and feel to file search

File: src/components/ProjectSearch.js [hunk 1]
```diff
@@ -118,8 +124,16 @@ export class ProjectSearch extends Component<Props, State> {
     }
   }
 
-  doSearch(searchTerm: string) {
-    this.props.searchSources(searchTerm);
+  getModifiers() {
```
Comment (darkwing) [inline src/components/ProjectSearch.js]: If you change this to `getModifiers = () => {  }`, you can avoid binding on line 98.

File: src/components/ProjectSearch.js [hunk 2]
```diff
@@ -167,7 +181,12 @@ export class ProjectSearch extends Component<Props, State> {
     this.setState({ focusedItem: null });
     const query = sanitizeQuery(this.state.inputValue);
     if (query) {
-      this.doSearch(query);
+      const modifiers = {
```
Comment (darkwing) [inline src/components/ProjectSearch.js]: I think you can use your `getModifiers` function here.

File: src/components/shared/SearchInput.css [hunk 3]
```diff
@@ -119,3 +119,15 @@
 .search-field.big .close-btn {
   margin-inline-end: 8px;
 }
+
+.search-field .search-nav-buttons .active {
+  background: #00000030;
```
Comment (darkwing) [inline src/components/shared/SearchInput.css]: Where did these colors come from?  You should be able to use the same colors as the other search field.

File: src/components/shared/SearchInput.js [hunk 4]
```diff
@@ -107,17 +116,74 @@ class SearchInput extends Component<Props, State> {
     return <AccessibleImage className="search" />;
   }
 
+  // handleModifierRegexMatch()
```
Comment (darkwing) [inline src/components/shared/SearchInput.js]: These comments need addressing.

File: src/components/shared/SearchInput.js [hunk 5]
```diff
@@ -209,11 +275,20 @@ class SearchInput extends Component<Props, State> {
   renderNav() {
     const { count, handleNext, handlePrev } = this.props;
     if ((!handleNext && !handlePrev) || (!count || count == 1)) {
-      return;
+      return (
+        <div className="search-nav-buttons">
+          <span style={{ padding: "5px" }}>Modifiers:</span>
```
Comment (darkwing) [inline src/components/shared/SearchInput.js]: We should use CSS for the padding to avoid duplication

**Comment by darkwing** [comment]
This is a really awesome start @charanjit-singh !  I left a few comments to continue you on the journey!

File: src/components/shared/SearchInput.css (hunk 3 above)
Comment (charanjit-singh) [reply src/components/shared/SearchInput.css]: Hi @darkwing !
From where Can I get the list of css variables ?
I'm unable to find this in repository.

File: src/components/shared/SearchInput.js (hunk 5 above)
Comment (charanjit-singh) [reply src/components/shared/SearchInput.js]: Okay

**Pull request closed without merging**